[Bug]: Unable to setup storybook (angular 20) Error: Unknown argument: silent

### Describe the bug

Hello. I'm trying to setup  storybook for a project: Angular 20 + Ionic 8. I'm running `npm create storybook@latest` and it fails at the end with `Error: Unknown argument: silent`.

Then I try to run it, and I get the error: 
```
ERROR in .storybook/preview.ts:1:30 - error TS2307: Cannot find module '@storybook/angular' or its corresponding type declarations.

1 import type { Preview } from '@storybook/angular'
                               ~~~~~~~~~~~~~~~~~~~~

ERROR in .storybook/preview.ts:2:33 - error TS2307: Cannot find module '@storybook/addon-docs/angular' or its corresponding type declarations.

2 import { setCompodocJson } from "@storybook/addon-docs/angular";
                                  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

ERROR in src/stories/button.stories.ts:1:37 - error TS2307: Cannot find module '@storybook/angular' or its corresponding type declarations.

1 import type { Meta, StoryObj } from '@storybook/angular';
                                      ~~~~~~~~~~~~~~~~~~~~

ERROR in src/stories/button.stories.ts:2:20 - error TS2307: Cannot find module 'storybook/test' or its corresponding type declarations.

2 import { fn } from 'storybook/test';
                     ~~~~~~~~~~~~~~~~

ERROR in src/stories/header.stories.ts:1:37 - error TS2307: Cannot find module '@storybook/angular' or its corresponding type declarations.

1 import type { Meta, StoryObj } from '@storybook/angular';
                                      ~~~~~~~~~~~~~~~~~~~~

ERROR in src/stories/header.stories.ts:2:20 - error TS2307: Cannot find module 'storybook/test' or its corresponding type declarations.

2 import { fn } from 'storybook/test';
                     ~~~~~~~~~~~~~~~~

ERROR in src/stories/page.stories.ts:1:37 - error TS2307: Cannot find module '@storybook/angular' or its corresponding type declarations.

1 import type { Meta, StoryObj } from '@storybook/angular';
                                      ~~~~~~~~~~~~~~~~~~~~

ERROR in src/stories/page.stories.ts:2:43 - error TS2307: Cannot find module 'storybook/test' or its corresponding type declarations.

2 import { expect, userEvent, within } from 'storybook/test';
                                            ~~~~~~~~~~~~~~~~
                               ~~~~~~~~~~~~~~~~~~~~
```

Commands:

```
 npm create storybook@latest

> appointments_client@0.0.1228 npx
> create-storybook


┌  Initializing Storybook
│
●  Adding Storybook version 10.2.16 to your project
│
◇  Framework detected: angular
│
◇  What configuration should we install?
│  Recommended: Component development and docs
│
│  Adding Storybook support to your "app" project
│
│  Compodoc is a great tool to generate documentation for your Angular projects.
│  Storybook can use the documentation generated by Compodoc to extract argument
│  definitions and JSDOC comments to display them in the Storybook UI. We highly
│  recommend using Compodoc for your Angular projects to get the best experience
│  out of Storybook.
│
◇  Do you want to use Compodoc for documentation?
│  Yes
│
◆  Storybook configuration generated
│
│  - Configuring ESLint plugin
│  - Configuring main.ts
│  - Configuring preview.ts
│  - Copying framework templates
│
◆  Dependencies added to package.json
│
│  Adding devDependencies:
│  - storybook@^10.2.16
│  - @storybook/angular@^10.2.16
│  - @storybook/addon-a11y@^10.2.16
│  - @storybook/addon-docs@^10.2.16
│  - @compodoc/compodoc@^1.2.1
│  - eslint-plugin-storybook@^10.2.16
│
◇  Dependencies installed
│
◆  Addons configured successfully
│
│  ✅ @storybook/addon-a11y
│  ✅ @storybook/addon-docs
│
◇  Storybook was successfully installed in your project!
│
│  To run Storybook manually, run ng run app:storybook. CTRL+C to stop.
│
│  Wanna know more about Storybook? Check out https://storybook.js.org/
│  Having trouble or want to chat? Join us at https://discord.gg/storybook/
│
└

Error: Unknown argument: silent
```

and then when running it:
```
ng run app:storybook  --debug-webpack

│
●  Starting...
│
●  Addon-docs: using MDX3
│
●  Using implicit CSS loaders
│
●  Using angular browser target options from "app:build"
│
●  Using angular project with "tsConfig:.storybook/tsconfig.json"
One or more browsers which are configured in the project's Browserslist configuration fall outside Angular's browser support for this version.
Unsupported browsers:
and_qq 14.9, and_uc 15.5, android 145, kaios 3.0-3.1, op_mob 80, opera 125, opera 124, samsung 29, samsung 28
│
●  Using default Webpack5 setup
<i> [webpack-dev-middleware] wait until bundle finished
ERROR in ./src/polyfills.ts
Module build failed (from ./node_modules/@ngtools/webpack/src/ivy/index.js):
Error: /Users/chicho/Documents/projects/appointia/appointments_client/src/polyfills.ts is missing from the TypeScript compilation. Please make sure it is in your tsconfig via the 'files' or 'include' property.
    at /Users/chicho/Documents/projects/appointia/appointments_client/node_modules/@ngtools/webpack/src/ivy/loader.js:75:26

ERROR in .storybook/preview.ts:1:30 - error TS2307: Cannot find module '@storybook/angular' or its corresponding type declarations.

1 import type { Preview } from '@storybook/angular'
                               ~~~~~~~~~~~~~~~~~~~~

ERROR in .storybook/preview.ts:2:33 - error TS2307: Cannot find module '@storybook/addon-docs/angular' or its corresponding type declarations.

2 import { setCompodocJson } from "@storybook/addon-docs/angular";
                                  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

ERROR in src/stories/button.stories.ts:1:37 - error TS2307: Cannot find module '@storybook/angular' or its corresponding type declarations.

1 import type { Meta, StoryObj } from '@storybook/angular';
                                      ~~~~~~~~~~~~~~~~~~~~

ERROR in src/stories/button.stories.ts:2:20 - error TS2307: Cannot find module 'storybook/test' or its corresponding type declarations.

2 import { fn } from 'storybook/test';
                     ~~~~~~~~~~~~~~~~

ERROR in src/stories/header.stories.ts:1:37 - error TS2307: Cannot find module '@storybook/angular' or its corresponding type declarations.

1 import type { Meta, StoryObj } from '@storybook/angular';
                                      ~~~~~~~~~~~~~~~~~~~~

ERROR in src/stories/header.stories.ts:2:20 - error TS2307: Cannot find module 'storybook/test' or its corresponding type declarations.

2 import { fn } from 'storybook/test';
                     ~~~~~~~~~~~~~~~~

ERROR in src/stories/page.stories.ts:1:37 - error TS2307: Cannot find module '@storybook/angular' or its corresponding type declarations.

1 import type { Meta, StoryObj } from '@storybook/angular';
                                      ~~~~~~~~~~~~~~~~~~~~

ERROR in src/stories/page.stories.ts:2:43 - error TS2307: Cannot find module 'storybook/test' or its corresponding type declarations.

2 import { expect, userEvent, within } from 'storybook/test';
                                            ~~~~~~~~~~~~~~~~

preview compiled with 9 errors
│
▲  Module Warning (from ./node_modules/sass-loader/dist/cjs.js):
│  Deprecation Warning on line 27, column 8 of
│  file:///Users/chicho/Documents/projects/appointia/appointments_client/src/global.scss:27:8:
│  Sass @import rules are deprecated and will be removed in Dart Sass 3.0.0.

│  More info and automated migrator: https://sass-lang.com/d/import

│  27 | @import './theme/appo-themes';


│  src/global.scss 28:9  root stylesheet

│
▲  Module Warning (from ./node_modules/sass-loader/dist/cjs.js):
│  Deprecation Warning on line 28, column 8 of
│  file:///Users/chicho/Documents/projects/appointia/appointments_client/src/global.scss:28:8:
│  Sass @import rules are deprecated and will be removed in Dart Sass 3.0.0.

│  More info and automated migrator: https://sass-lang.com/d/import

│  28 | @import './theme/appo-ionic-overrides/appo-ionic-index';


│  src/global.scss 29:9  root stylesheet

│
▲  Module Warning (from ./node_modules/sass-loader/dist/cjs.js):
│  Deprecation Warning on line 30, column 8 of
│  file:///Users/chicho/Documents/projects/appointia/appointments_client/src/global.scss:30:8:
│  Sass @import rules are deprecated and will be removed in Dart Sass 3.0.0.

│  More info and automated migrator: https://sass-lang.com/d/import

│  30 | @import './theme/marker';


│  src/global.scss 31:9  root stylesheet

│
▲  Module Warning (from ./node_modules/sass-loader/dist/cjs.js):
│  Deprecation Warning on line 31, column 8 of
│  file:///Users/chicho/Documents/projects/appointia/appointments_client/src/global.scss:31:8:
│  Sass @import rules are deprecated and will be removed in Dart Sass 3.0.0.

│  More info and automated migrator: https://sass-lang.com/d/import

│  31 | @import './theme/mapbox-geocoder';


│  src/global.scss 32:9  root stylesheet

│
▲  Module Warning (from ./node_modules/sass-loader/dist/cjs.js):
│  Deprecation Warning on line 32, column 8 of
│  file:///Users/chicho/Documents/projects/appointia/appointments_client/src/global.scss:32:8:
│  Sass @import rules are deprecated and will be removed in Dart Sass 3.0.0.

│  More info and automated migrator: https://sass-lang.com/d/import

│  32 | @import './theme/spot-options';


│  src/global.scss 33:9  root stylesheet

│
▲  Module Warning (from ./node_modules/sass-loader/dist/cjs.js):
│  31 repetitive deprecation warnings omitted.
│  Run in verbose mode to see all warnings.
│
▲  /Users/chicho/Documents/projects/appointia/appointments_client/node_modules/@storybook/addon-a11y/dist/_browser-chunks/chunk-2FVV6YOY.js
│  depends on 'axe-core'. CommonJS or AMD dependencies can cause optimization
│  bailouts.
│  For more info see: https://angular.dev/tools/cli/build#configuring-commonjs-d...
│
▲  /Users/chicho/Documents/projects/appointia/appointments_client/node_modules/storybook/dist/manager-api/index.js
│  depends on 'semver'. CommonJS or AMD dependencies can cause optimization
│  bailouts.
│  For more info see: https://angular.dev/tools/cli/build#configuring-commonjs-d...
│
▲  /Users/chicho/Documents/projects/appointia/appointments_client/node_modules/webpack-hot-middleware/client.js?reload=true&quiet=false&overlay={"errors":true,"warnings":false,"runtimeErrors":false}&noInfo=false
│  depends on 'strip-ansi'. CommonJS or AMD dependencies can cause optimization
│  bailouts.
│  For more info see: https://angular.dev/tools/cli/build#configuring-commonjs-d...
│
▲  DefinePlugin
│  Conflicting values for 'process.env.NODE_ENV'
│
■  Failed to build the preview
99% end closing watch compilation│
▲  Force closed preview build
│
■  SB_BUILDER-WEBPACK5_0003 (WebpackCompilationError): There were problems when
│  compiling your code with Webpack.
│  Run Storybook with --debug-webpack for more information.
│  at starter
│  (file://./node_modules/@storybook/builder-webpack5/dist/index.js:799:11)
│  at starter.next (<anonymous>)
│  at Module.start
│  (file://./node_modules/@storybook/builder-webpack5/dist/index.js:857:34)
│  at process.processTicksAndRejections (node:internal/process/task_queues:105:5)
│  at async storybookDevServer
│  (file://./node_modules/storybook/dist/core-server/index.js:7295:83)
│  at async buildOrThrow
│  (file://./node_modules/storybook/dist/core-server/index.js:4506:12)
│  at async buildDevStandalone
│  (file://./node_modules/storybook/dist/core-server/index.js:7710:41)
│  at async withTelemetry
│  (file://./node_modules/storybook/dist/_node-chunks/chunk-CSGMAQQ4.js:218:12)
│  at async runInstance
│  (file://./node_modules/@storybook/angular/dist/builders/start-storybook/index.js:141:20)
│  at async
│  file://./node_modules/@storybook/angular/dist/builders/start-storybook/index.js:114:24
An unhandled exception occurred: Broken build, fix the error above.
You may need to refresh the browser.
See "/private/var/folders/2b/30ygqrl15szbt3g9f3dvzy0r0000gn/T/ng-5OZ3fU/angular-errors.log" for further details.

```

Full log [link](https://drive.google.com/file/d/14qrGF3Yq0DWywM0djxQe5VJGqJ6r-3OG/view?usp=sharing)

Dependencies

```
  "dependencies": {
    "@algolia/client-search": "^5.49.1",
    "@andypf/json-viewer": "^2.2.0",
    "@angular/cdk": "^20.2.11",
    "@angular/common": "^20.3.9",
    "@angular/core": "^20.3.9",
    "@angular/forms": "^20.3.9",
    "@angular/platform-browser": "^20.3.9",
    "@angular/platform-browser-dynamic": "^20.3.9",
    "@angular/router": "^20.3.9",
    "@angular/service-worker": "^20.3.9",
    "@apollo/client": "^3.13.8",
    "@aws-sdk/abort-controller": "^3.374.0",
    "@aws-sdk/client-appsync": "^3.758.0",
    "@aws-sdk/client-cognito-identity": "^3.758.0",
    "@aws-sdk/credential-provider-cognito-identity": "^3.758.0",
    "@capacitor-community/contacts": "^7.1.0",
    "@capacitor-firebase/analytics": "^8.0.1",
    "@capacitor-firebase/authentication": "^8.0.1",
    "@capacitor-firebase/crashlytics": "^8.0.1",
    "@capacitor/android": "^8.2.0",
    "@capacitor/app": "^8.0.1",
    "@capacitor/app-launcher": "^8.0.1",
    "@capacitor/browser": "^8.0.2",
    "@capacitor/camera": "^8.0.2",
    "@capacitor/clipboard": "^8.0.1",
    "@capacitor/core": "^8.2.0",
    "@capacitor/device": "^8.0.1",
    "@capacitor/geolocation": "^8.1.0",
    "@capacitor/haptics": "^8.0.1",
    "@capacitor/ios": "^8.2.0",
    "@capacitor/keyboard": "^8.0.1",
    "@capacitor/local-notifications": "^8.0.2",
    "@capacitor/preferences": "^8.0.1",
    "@capacitor/push-notifications": "^8.0.2",
    "@capacitor/share": "^8.0.1",
    "@capacitor/splash-screen": "^8.0.1",
    "@capacitor/status-bar": "^8.0.1",
    "@capawesome/capacitor-android-edge-to-edge-support": "^8.0.6",
    "@capawesome/capacitor-app-update": "^8.0.3",
    "@ebarooni/capacitor-calendar": "^8.0.1",
    "@ionic/angular": "^8.8.0",
    "@ionic/pwa-elements": "^3.3.0",
    "@mapbox/mapbox-gl-geocoder": "^5.0.1",
    "@ngrx/component-store": "^20.0.0",
    "@ngrx/effects": "^20.0.0",
    "@ngrx/entity": "^20.0.0",
    "@ngrx/operators": "^20.0.0",
    "@ngrx/router-store": "^20.0.0",
    "@ngrx/schematics": "^20.0.0",
    "@ngrx/signals": "^20.0.0",
    "@ngrx/store": "^20.0.0",
    "@ngrx/store-devtools": "^20.0.0",
    "@sentry/angular": "^9.30.0",
    "@sentry/browser": "^9.4.0",
    "@swc/core": "^1.15.18",
    "@swc/jest": "^0.2.39",
    "@types/mapbox__mapbox-gl-geocoder": "^4.7.3",
    "@types/mapbox-gl": "^1.13.4",
    "algoliasearch": "^5.49.1",
    "apollo-angular": "^10.0.3",
    "apollo-codegen": "^0.20.2",
    "aws-appsync": "^4.1.9",
    "aws-appsync-auth-link": "^3.0.7",
    "backoff-rxjs": "^6.5.7",
    "bowser": "^2.12.0",
    "compare-versions": "^6.1.1",
    "countries-and-timezones": "^3.8.0",
    "cropperjs": "^1.6.2",
    "crypto-browserify": "^3.12.0",
    "date-fns": "^2.30.0",
    "easyqrcodejs": "^4.6.1",
    "escape-string-regexp": "^5.0.0",
    "fastdom": "^1.0.12",
    "firebase": "^10.14.1",
    "flexsearch": "^0.8.212",
    "graphql": "15.8.0",
    "install": "^0.13.0",
    "ionicons": "^8.0.13",
    "is-url": "^1.2.4",
    "js-sha256": "^0.9.0",
    "jwt-decode": "^3.1.2",
    "ksuid": "^1.2.0",
    "libphonenumber-js": "^1.12.38",
    "lodash": "^4.17.23",
    "lodash-es": "^4.17.23",
    "lottie-web": "^5.13.0",
    "mapbox-gl": "^1.13.3",
    "moment-timezone": "^0.5.41",
    "ngx-json-viewer": "^2.4.0",
    "ngx-lottie": "^11.0.1",
    "npm": "^11.6.4",
    "posthog-js": "^1.359.1",
    "redux": "^5.0.1",
    "rxjs": "~7.8.2",
    "swiper": "^11.2.10",
    "tslib": "^2.8.1",
    "util": "^0.12.5"
  },
  "devDependencies": {
    "@angular-builders/custom-webpack": "^19.0.1",
    "@angular-devkit/architect": "^0.2003.8",
    "@angular-devkit/build-angular": "^20.3.8",
    "@angular-devkit/core": "^20.3.8",
    "@angular-devkit/schematics": "^20.3.8",
    "@angular-eslint/schematics": "^20.1.1",
    "@angular-eslint/template-parser": "^20.1.1",
    "@angular/cli": "^20.3.8",
    "@angular/compiler": "^20.3.9",
    "@angular/compiler-cli": "^20.3.9",
    "@angular/language-service": "^20.3.9",
    "@babel/runtime": "^7.26.9",
    "@capacitor/assets": "^3.0.5",
    "@capacitor/cli": "^8.0.2",
    "@ionic/angular-toolkit": "^12.3.0",
    "@playwright/test": "^1.58.2",
    "@rrweb/types": "^2.0.0-alpha.16",
    "@sentry/cli": "^2.57.0",
    "@types/algoliasearch": "^4.0.4",
    "@types/base-64": "^1.0.2",
    "@types/is-url": "^1.2.30",
    "@types/jest": "^30.0.0",
    "@types/lodash": "^4.14.202",
    "@types/node": "^20.9.0",
    "@typescript-eslint/eslint-plugin": "^7.13.0",
    "@typescript-eslint/parser": "^7.13.0",
    "@typescript-eslint/utils": "^7.16.1",
    "codelyzer": "^6.0.2",
    "eslint-config-prettier": "^9.1.0",
    "eslint-plugin-prettier": "^5.2.1",
    "eslint-plugin-unused-imports": "^4.0.0",
    "jest": "^30.2.0",
    "jest-preset-angular": "^14.6.0",
    "prettier": "^3.5.3",
    "prettier-eslint": "^16.3.0",
    "protractor": "~7.0.0",
    "pwa-asset-generator": "^2.3.0",
    "rrweb-snapshot": "^2.0.0-alpha.16",
    "ts-node": "~10.9.2",
    "tslint": "~6.1.3",
    "typescript": "~5.8.2",
    "webpack-bundle-analyzer": "^4.7.0",
    "storybook": "^10.2.16",
    "@storybook/angular": "^10.2.16",
    "@storybook/addon-a11y": "^10.2.16",
    "@storybook/addon-docs": "^10.2.16",
    "@compodoc/compodoc": "^1.2.1",
    "eslint-plugin-storybook": "^10.2.16"
  },
  "type": "module"
```



### Reproduction link

None

### Reproduction steps

Run `npm create storybook@latest`

### System

```bash
│
│  Storybook Environment Info:
│

│  System:
│  OS: macOS 26.3.1
│  CPU: (14) arm64 Apple M4 Pro
│  Shell: 5.9 - /bin/zsh
│  Binaries:
│  Node: 24.2.0 - /Users/chicho/.nvm/versions/node/v24.2.0/bin/node
│  npm: 11.11.0 -
│  /Users/chicho/Documents/projects/appointia/appointments_client/node_modules/.bin/npm
│  <----- active
│  pnpm: 10.17.0 - /Users/chicho/Library/pnpm/pnpm
│  Browsers:
│  Chrome: 145.0.7632.160
│  Firefox: 145.0.2
│  Safari: 26.3.1
│  npmPackages:
│  @storybook/addon-a11y: ^10.2.16 => 10.2.16
│  @storybook/addon-docs: ^10.2.16 => 10.2.16
│  @storybook/angular: ^10.2.16 => 10.2.16
│  eslint-plugin-storybook: ^10.2.16 => 10.2.16
│  storybook: ^10.2.16 => 10.2.16
```

### Additional context

_No response_

@dosu

i just did that and I still get:
```
Using default Webpack5 setup
<i> [webpack-dev-middleware] wait until bundle finished
ERROR in .storybook/preview.ts:1:30 - error TS2307: Cannot find module '@storybook/angular' or its corresponding type declarations.

1 import type { Preview } from '@storybook/angular'
                               ~~~~~~~~~~~~~~~~~~~~

ERROR in .storybook/preview.ts:2:33 - error TS2307: Cannot find module '@storybook/addon-docs/angular' or its corresponding type declarations.

2 import { setCompodocJson } from "@storybook/addon-docs/angular";
                                  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

ERROR in src/stories/button.stories.ts:1:37 - error TS2307: Cannot find module '@storybook/angular' or its corresponding type declarations.

1 import type { Meta, StoryObj } from '@storybook/angular';
                                      ~~~~~~~~~~~~~~~~~~~~

ERROR in src/stories/button.stories.ts:2:20 - error TS2307: Cannot find module 'storybook/test' or its corresponding type declarations.

2 import { fn } from 'storybook/test';
                     ~~~~~~~~~~~~~~~~

ERROR in src/stories/header.stories.ts:1:37 - error TS2307: Cannot find module '@storybook/angular' or its corresponding type declarations.

1 import type { Meta, StoryObj } from '@storybook/angular';
                                      ~~~~~~~~~~~~~~~~~~~~

ERROR in src/stories/header.stories.ts:2:20 - error TS2307: Cannot find module 'storybook/test' or its corresponding type declarations.

2 import { fn } from 'storybook/test';
                     ~~~~~~~~~~~~~~~~

ERROR in src/stories/page.stories.ts:1:37 - error TS2307: Cannot find module '@storybook/angular' or its corresponding type declarations.

1 import type { Meta, StoryObj } from '@storybook/angular';
                                      ~~~~~~~~~~~~~~~~~~~~

ERROR in src/stories/page.stories.ts:2:43 - error TS2307: Cannot find module 'storybook/test' or its corresponding type declarations.

2 import { expect, userEvent, within } from 'storybook/test';
                                            ~~~~~~~~~~~~~~~~

preview compiled with 8 errors
```

--- a/code/lib/create-storybook/templates/angular/application/template-csf/.storybook/tsconfig.json
+++ b/code/lib/create-storybook/templates/angular/application/template-csf/.storybook/tsconfig.json
@@ -3,7 +3,8 @@
   "compilerOptions": {
     "types": ["node"],
     "allowSyntheticDefaultImports": true,
-    "resolveJsonModule": true
+    "resolveJsonModule": true,
+    "noCheck": true
   },
   "exclude": ["../src/test.ts", "../src/**/*.spec.ts"],
   "include": ["../src/**/*.stories.*", "./preview.ts"],

--- a/code/lib/create-storybook/templates/angular/application/template-csf/.storybook/tsconfig.json
+++ b/code/lib/create-storybook/templates/angular/application/template-csf/.storybook/tsconfig.json
@@ -4,7 +4,7 @@
     "types": ["node"],
     "allowSyntheticDefaultImports": true,
     "resolveJsonModule": true,
-    "noCheck": true
+    "moduleResolution": "bundler"
   },
   "exclude": ["../src/test.ts", "../src/**/*.spec.ts"],
   "include": ["../src/**/*.stories.*", "./preview.ts"],

